src/layouts/components/searchbar/SearchResult.tsx

```diff
@@ -0,0 +1,112 @@
+import { Combobox } from "@headlessui/react";
+import { Fragment } from "react";
+import clsx from "clsx";
+import { FiBox } from "react-icons/fi";
+import { IoCloseCircleSharp } from "react-icons/io5";
+import { TbLoaderQuarter } from "react-icons/tb";
+import { Link } from "@components/commons/Link";
+import { TransactionsIcon } from "@components/icons/Transactions";
+
+export interface SearchResult {
+  url: string;
+  title: string;
+  type: string;
+}
+
+export function SearchResultTable({
+  searchResults,
+  isSearching,
+}: {
+  searchResults?: SearchResult[];
+  isSearching: boolean;
+}): JSX.Element {
+  if (isSearching) {
+    return <SearchingMessage />;
+  }
+
+  if (searchResults === undefined || searchResults.length === 0) {
+    return <NoResultsMessage />;
+  }
+
+  return (
+    <Combobox.Options className="focus:outline-none" static>
+      {searchResults.map((searchResult) => (
+        <SearchResultRow
+          searchResults={searchResult}
+          key={`${searchResult.type}.${searchResult.title}`}
+        />
+      ))}
+    </Combobox.Options>
+  );
+}
+
+const iconMapping = {
```

Comment:
instead of string mapping, can we use `Record<T>` ? ref: https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type